client/checkout/classic/event-handlers.js

```diff
@@ -267,16 +277,35 @@ jQuery( function ( $ ) {
 			SHORTCODE_BILLING_ADDRESS_FIELDS.address_1,
 			SHORTCODE_BILLING_ADDRESS_FIELDS.city,
 			SHORTCODE_BILLING_ADDRESS_FIELDS.postcode,
-		].filter( ( field ) => billingFieldsDisplayed.includes( field ) );
+		].filter( ( field ) => enabledBillingFields[ field ] );
+
+		const country = billingFieldsToValidate.includes(
+			SHORTCODE_BILLING_ADDRESS_FIELDS.country
+		)
+			? document.querySelector(
+					`#${ SHORTCODE_BILLING_ADDRESS_FIELDS.country }`
+			  )?.value
+			: null;
 
 		// We need to just find one field with missing information. If even only one is missing, just return early.
 		return Boolean(
 			billingFieldsToValidate.find( ( fieldName ) => {
 				const $field = document.querySelector( `#${ fieldName }` );
-				const $formRow = $field.closest( '.form-row' );
-				const isRequired = $formRow.classList.contains(
-					'validate-required'
-				);
+				let isRequired = false;
+
+				if ( country && locale && fieldName !== 'billing_email' ) {
+					const key = fieldName.replace( 'billing_', '' );
+					const fieldSettings = jQuery.extend(
+						true,
+						{},
+						locale.default[ key ],
+						locale[ country ][ key ]
+					);
+					isRequired = fieldSettings.required;
+				} else {
+					isRequired = enabledBillingFields[ fieldName ]?.required;
+				}
```

Comment:
The initialization of `isRequired` could be simplified like this, since it'll always be overwritten by the `else` statement:
```suggestion
				let isRequired = enabledBillingFields[ fieldName ]?.required;

				if ( country && locale && fieldName !== 'billing_email' ) {
					const key = fieldName.replace( 'billing_', '' );
					const fieldSettings = jQuery.extend(
						true,
						{},
						locale.default[ key ],
						locale[ country ][ key ]
					);
					isRequired = fieldSettings.required;
				}
```